nx-X11/programs/Xserver/hw/nxagent/Args.h

```diff
@@ -53,7 +53,7 @@ struct UserGeometry{
 extern Bool nxagentUseNXTrans;
 
 extern char nxagentSessionId[];
-extern char nxagentDisplayName[];
+extern char nxagentDisplayName[1024];
```

Comment:
Hmm, I had to add the size here because otherwise the sizeof would fail if used in sources _other than_ Args.c. So  I am wondering if you are really right about the "only a hint" stuff.

Generally I want to get rid of all those limits as they are only artificial. So I will change the code one day to not use fixed array.

Reply:
Looks like I've been wrong - `T foo[];` declares an incomplete type that must be properly defined at a later point (unless the code in question does not need storage information).

Multiple declarations for he same object with different sizes lead a compile error in recent versions of both clang and gcc.

There is a special case that I confused this with: within parameters declared like this, the size is merely a hint:

```
b.c: In function ‘get_size_2’:
b.c:9:18: warning: ‘sizeof’ on array function parameter ‘arg’ will return size of ‘char *’ [-Wsizeof-array-argument]
   return (sizeof (arg));
                  ^
b.c:8:25: note: declared here
 size_t get_size_2 (char arg[6]) {
                         ^~~

(the array I pass in actually is `char foo[5]` - and that works)
```

For now, I'd really use a macro.

Bounding even pure pointers would also be a good idea, even if that adds an artificial limit. We can certainly raise this limit, but not doing it at all is dangerous in case the input data is invalid.

Reply:
ok, I have added macros (for now). And fixed two bugs in nxagentGetDialogName: the wrong string was terminated and there where no length checks.